alchemy/src/vercel/project.ts

```diff
@@ -0,0 +1,418 @@
+            const deleteResponse = await api.delete(
+              `/projects/${this.output.id}`,
+            );
+
+            if (!deleteResponse.ok && deleteResponse.status !== 404) {
+              console.error(
+                "Error deleting project:",
+                deleteResponse.statusText,
+              );
+            }
+          }
+        } catch (error) {
+          console.error("Error deleting project:", error);
+        }
+
+        return this.destroy();
+      }
+
+      case "update": {
+        const api = await createVercelApi({
+          baseUrl: "https://api.vercel.com/v9",
+          accessToken,
+        });
+
+        if (!this.output?.id) {
+          throw new Error("Cannot update project without ID");
+        }
+
+        // 409 Conflict: Can't update name, so remove it from the props
+        // 400 Invalid Request: Should NOT have additional property `environmentVariables`
+        const { name, environmentVariables, ...rest } = props;
+
+        const response = await api.patch(`/projects/${this.output.id}`, rest);
+        const data = (await response.json()) as {
+          id: string;
+          accountId: string;
+          createdAt: number;
+          updatedAt: number;
+          latestDeployment?: {
+            id: string;
+            url: string;
+          };
+        };
+
+        if (environmentVariables) {
+          const envApi = await createVercelApi({
+            baseUrl: "https://api.vercel.com/v10",
+            accessToken,
+          });
+
+          await envApi.post(
+            `/projects/${this.output.id}/env?upsert=true`,
+            environmentVariables.map((envVar) => ({
+              ...envVar,
+              value:
+                envVar.type === "encrypted"
+                  ? envVar.value.unencrypted
+                  : envVar.value,
+            })),
+          );
+
+          // Find previous env vars that are not in the new list and delete them
+          const { envs } = (await envApi
+            .get(`/projects/${this.output.id}/env`)
+            .then((res) => res.json())) as {
+            envs: Array<{ id: string; key: string }>;
+          };
+          for (const previousEnv of envs) {
+            if (
+              environmentVariables.some(
+                (envVar) => envVar.key === previousEnv.key,
+              )
+            ) {
+              continue;
+            }
+
+            await envApi.delete(
+              `/projects/${this.output.id}/env/${previousEnv.id}`,
+            );
+          }
+        }
+
+        return this({
+          id: data.id,
+          accountId: data.accountId,
+          createdAt: data.createdAt,
+          updatedAt: data.updatedAt,
+          latestDeployment: data.latestDeployment,
+          ...props,
+        });
+      }
+
+      default: {
+        const api = await createVercelApi({
+          baseUrl: "https://api.vercel.com/v11",
+          accessToken,
+        });
```

Comment:
redundant

Reply:
So far this file has `v9`, `v10` and `v11` APIs. These can go into the `deleteProject` helpers.